File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if(i == "[" or i == "(" or i == "{"):
                stack.append(i)
            elif(i == ')' and len(stack) >0):
                if(stack.pop()!="("):
                    return False
            elif(i == ']' and len(stack) >0):
                if(stack.pop()!="["):
                    return False
            elif(i == '}' and len(stack) >0):
                if(stack.pop()!="{"):
                    return False   
            else:
                return False
        return True if len(stack)==0 else False
                
        

File: 0020-valid-parentheses/0020-valid-parentheses.kt
class Solution {
fun isValid(s: String): Boolean {
    var stack = Stack<Char>()
    
    s.toCharArray().forEach { 
        if(it == '(' || it == '{' || it == '['){
            stack.push(it)
        }else{
            if(stack.isEmpty()) return false
            val top = stack.pop()
            if(it == ')' && top != '(') return false
            if(it == '}' && top != '{') return false
            if(it == ']' && top != '[') return false
        }
    }
    return stack.isEmpty()
}
}